### ISSUE TITLE
Optionally override DisplayVersion

### ISSUE THREAD

**Issue by o-l-a-v** [microsoft/winget-cli#2294]
### Description of the new feature / enhancement

Some installers do not write correct DisplayVersion to registry, like VLC (until next stable version).

It would be great if Winget could implement a way to override default behavior. This would make Winget more useful, practical and resilient.

Ideally this should be handled correctly by the developers of the program in question, but it's just not practical or realistic to think like that. Take VLC for instance, this has been a known issue for months, but they still haven't published a fixed version of the installer. Unless users write correct DisplayVersion to registry themselves, VLC gets reinstalled every time one run 'winget upgrade -all'. Better to implement an optional override, than to wait for a fix. In some cases other than VLC, the devs might not even bother to fix it.

### Proposed technical implementation details

I'm thinking about two potential ways of doing this.

1. Override the registry key DisplayVersion after successfull install, by setting the registry value to match what the manifest says.
2. Override detection method to check something else, for instance file version of a path.
  * This would work for VLC for instance.

**Comment by denelon** [comment]
@o-l-a-v 

I'm not sure we want to make registry changes for installers. This could have a cascading effect of unintended consequences. There are also packages that update themselves, and users may also download or install upgrades/changes through other means. We're looking at these types of issues as they relate to catalog health. I'll keep this issue open for now so we can continue to evaluate what to do with this class of installer related issues.

**Comment by o-l-a-v** [comment]
I agree, thus optional override for version detection method might be something to look into?

**Comment by giggio** [comment]
I created a duplicate of this issue, so I'll try to add a little bit of context to help make it more searcheable.
To me, the issue surfaced as the `unknown` version showing on `winget list`, and also when using `--include-unknown` when upgrading, which sometimes starts an upgrade process that was actually not needed.
Using the manifest version is, I believe, the correct approach here. The `--include-unknown` option could then be removed.

**Comment by denelon** [comment]
@giggio we can't trust that the application hasn't been updated independently from winget. That means if the version isn't being reported correctly, we can't reason about what is installed. We certainly hope that over time, installers improve, and packages are properly reporting their versions which gives a better user experience. That's why the packages that don't report their versions were excluded from "winget upgrade" and "winget upgrade --all". The user can attempt to upgrade any package individually, and if they are willing to go through a potentially painful batch of upgrades the "--include-unknown" can be used in conjunction with "--all".

**Comment by giggio** [comment]
@denelon This reasoning makes to me, but I believe we could find a way to respect that but also help the user.

What if winget captured the version from the manifest and reported with an `*`, and did not upgrade it automatically? Maybe still keep the upgrade of such installs under a flag, but allow the user to upgrade if an app seems to be outdated.

My use case is to use winget for all the upgrades, so if winget is reporting that it is (or isn't) outdated, then I can rely on that information. The current way does not help me at all, I don't know what version was installed unless I manually open the apps to check.

**Comment by JustMyGithub** [comment]
> @giggio we can't trust that the application hasn't been updated independently from winget. That means if the version isn't being reported correctly, we can't reason about what is installed. 

You are absolutely right, but what would be the problematic implication of this? 
1) proper update by the application/user: DisplayVersion would be updated, winget does not report or do anything
2a) improper update by the application/user: DisplayVersion likely does not exist, as the update did rewrite the uninstall key
2b) 2a, but DisplayVersion is the winget value: winget does overinstall the same version once, updates DisplayVersion as before and everything is fine

Consider tools that have not been updated for years. They will be overinstalled by winget again and again and as the installers are never updated, there will never be proper DisplayVersion value
